flang/test/Lower/OpenMP/copyprivate.f90

```diff
@@ -60,18 +60,18 @@ subroutine test_tp()
 
 !CHECK-LABEL: func @_QPtest_scalar
 !CHECK:         omp.parallel
-!CHECK:           %[[I1:.*]]:2 = hlfir.declare %{{.*}} {uniq_name = "_QFtest_scalarEi1"} : (!fir.ref<i32>) -> (!fir.ref<i32>, !fir.ref<i32>)
```

Comment:
It looks like most of these changes is reversing the original order of the clauses. Are these object lists already reversed in the output of `buildConstructQueue`? Or do you know where that's happening?

Reply:
It's getting reordered in re-composition.  The DSA clauses get collapsed, i.e. `private(x) private(y)` is replaced with `private(x, y)`, or `private(y, x)`, the order there is the result of how the implementation handles the clauses---there is no predefined order.

Maybe we should sort object lists in clauses that have them...

Reply:
I changed the recomposition code to try harder to maintain predictable order, and the number of modified tests is down to 3.